services/ui-src/src/components/modals/AddEditOverlayEntityModal.tsx

```diff
@@ -168,8 +169,13 @@ export const AddEditOverlayEntityModal = ({
     selectedEntity?.id
       ? (title = verbiage.addEditModalEditTitle)
       : (title = verbiage.addEditModalAddTitle);
-    if (entityName) {
-      title += entityName;
+    if (report?.reportType === ReportType.WP && entityName) {
+      title = `${title} ${entityName}`;
+    } else if (
+      report?.reportType === ReportType.SAR &&
+      selectedEntity?.objectiveProgress_objectiveName
+    ) {
+      title = `${title} ${selectedEntity?.objectiveProgress_objectiveName}`;
```

Comment:
non-blocking: technically you don't need the `?` on objective name since it must exist to pass the `if` condition

Reply:
true! removed it :) thank you.